service/routes.py

```diff
@@ -91,14 +92,37 @@ def create_inventory_records():
     statement += f"and condition: {inventory.condition} created."
     app.logger.info(statement)
     # return jsonify(inventory.serialize()), status.HTTP_201_CREATED
+
     return jsonify(inventory.serialize()), status.HTTP_201_CREATED, {"Location": location_url}
 
 
 @app.route("/inventory", methods=["GET"])
 def list_inventory_records():
     """Returns all of the Inventory records"""
-    app.logger.info("Request list of inventory records")
-    records = Inventory.all()
+    records = []
+    name = request.args.get("name")
+    condition = request.args.get("condition")
+    quantity = request.args.get("quantity")
+    active = request.args.get("active")
+
+    if name:
+        app.logger.info("Filtering by name: %s", name)
+        records = Inventory.find_by_name(name)
+    elif condition:
+        app.logger.info("Filtering by condition:%s", condition)
+        condition = Inventory.Condition(condition)
+        records = Inventory.find_by_condition(condition)
+    elif quantity:
+        app.logger.info("Filtering by quantity: %s", quantity)
+        records = Inventory.find_by_quantity(quantity)
+    elif active:
+        app.logger.info("Filtering by available: %s", active)
+        is_active = active.lower() in ["yes", "y", "true", "t", "1"]
```

Comment:
Can we have boolean here for active instead of string?

Reply:
we can have a boolean but ideally when the inventory manager types it, it will be a string or char format and not taken as a boolean so we have left it as a boolean here

Reply:
What if the user types in `T`/`True`/`Yes`/`Y`/`Yes!`?

Maybe we should move the responsibility of passing consistent data to frontend/ui rather than handling it at the backend.

Reply:
ohh hmm that's a good point. prof did a similar thing as well so I assumed this would have to be handled by us.

Reply:
for now we can leave it in the backend and then once we implement the ui we can update it here. what do you think?

Reply:
so that when we are testing our code at the end of the sprint on postman this feature works

Reply:
Maybe do it here itself, so that when we work on frontend/ui we have one less dependency to worry about.

Reply:
okie

Reply:
done